apps/passport-client/src/dispatch.ts

```diff
@@ -756,6 +770,12 @@ async function doSync(
     state.pcds,
     state.subscriptions
   );
+  const validationErrors = validateUpload(state.self, state.pcds);
+  if (validationErrors.length > 0) {
+    logAndUploadValidationErrors(validationErrors);
+    // TODO: what's the proper thing to do here?
```

Comment:
Certainly don't upload.  Probably put the user into an invalid state by setting state.userInvalid?  Reporting to the user of what actually happened will need tweaking, though.